server/src/auth.middleware.ts

```diff
@@ -15,15 +20,52 @@ export class AuthMiddleware implements NestMiddleware {
 
     req.session = false;
 
-    const { authorization = '' } = req.headers;
+    const { authorization = '', referer } = req.headers;
     const token = authorization.split(' ')[1];
 
     try {
-      req.session = await this.authService.validateCurrentToken(token);
+      // this promise will throw if invalid
+      let validatedToken = await this.authService.validateCurrentToken(token);
+
+      // the referer is the domain issuing the request. in this case, it'll be the client domain,
+      if (referer) {
+        // which includes the query params
+        const refererParams = new URL(referer);
+        const email = refererParams.searchParams.get('email'); // the query param, email, sent from the client. this is who the "Creeper" wants to be.
+        const { mail } = validatedToken; // the "creepers" actual email, for verification.
```

Comment:
can we rename this to something like "authedEmail"

Reply:
@godfreyyeung not really but we can alias it.

Reply:
it's named as such because it's provided by NYC ID

Reply:
ohh gotcha

Reply:
I think `{ mail: authedMail }` would be great, just for code clarity